packages/sitecore-jss-angular/src/components/placeholder.component.spec.ts

```diff
@@ -844,3 +897,320 @@ describe('SXA components', () => {
+      );
+      expect(elements[3].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="placeholder"></code>'
+      );
+    })
+  );
+
+  it('should render code blocks even if placeholder is empty', () =>
+    waitForAsync(async () => {
+      const component = layoutDataWithEmptyPlaceholder.sitecore.route;
+      const phKey = 'main';
+      comp.name = phKey;
+      comp.rendering = (component as unknown) as ComponentRendering;
+
+      fixture.detectChanges();
+      await fixture.whenStable();
+      fixture.detectChanges();
+      const elements = de.queryAll(By.css('code'));
+
+      expect(de.query(By.css('div.sc-jss-empty-placeholder'))).toBeDefined();
+
+      expect(elements.length).toBe(4);
+      expect(elements[0].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="open" chrometype="placeholder" id="main_00000000-0000-0000-0000-000000000000"></code>'
+      );
+      expect(elements[1].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="open" chrometype="rendering" id="nested123"></code>'
+      );
+      expect(elements[2].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="rendering"></code>'
+      );
+      expect(elements[3].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="placeholder"></code>'
+      );
+    }));
+
+  it(
+    'should render missing component with code blocks if component is not registered',
+    waitForAsync(async () => {
+      const component = layoutDataWithUnknownComponent.sitecore.route;
+      const phKey = 'main';
+      comp.name = phKey;
+      comp.rendering = (component as unknown) as ComponentRendering;
+
+      fixture.detectChanges();
+      await fixture.whenStable();
+      fixture.detectChanges();
+
+      expect(de.children.length).toBe(1);
+      const elements = de.queryAll(By.css('code'));
+
+      expect(elements.length).toBe(4);
+      expect(elements[0].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="open" chrometype="placeholder" id="main_00000000-0000-0000-0000-000000000000"></code>'
+      );
+      expect(elements[1].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="open" chrometype="rendering" id="nested123"></code>'
+      );
+      expect(elements[2].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="rendering"></code>'
+      );
+      expect(elements[3].nativeElement.outerHTML).toBe(
+        '<code type="text/sitecore" class="scpm" kind="close" chrometype="placeholder"></code>'
+      );
+      expect(de.query(By.directive(MissingComponentComponent))).toBeDefined();
+    })
+  );
+});
+
+describe('Placeholder Metadata: dynamic placeholder:', () => {
```

Comment:
```suggestion
describe('Placeholder Metadata: dynamic placeholder', () => {
```

Reply:
This is intentional. The way angular ng-test outputs unit test descriptions is `<Describe section> <it section>` instead of a more intuitive way of doing "describe" and "it" sections on two different lines. Having a colon here improves readability of test descriptions.